bundles/org.openhab.core.semantics/src/main/java/org/openhab/core/semantics/internal/SemanticTagRegistryImpl.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.semantics.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.common.registry.AbstractRegistry;
+import org.openhab.core.semantics.ManagedSemanticTagProvider;
+import org.openhab.core.semantics.SemanticTag;
+import org.openhab.core.semantics.SemanticTagProvider;
+import org.openhab.core.semantics.SemanticTagRegistry;
+import org.openhab.core.semantics.SemanticTags;
+import org.openhab.core.semantics.Tag;
+import org.openhab.core.semantics.TagInfo;
+import org.openhab.core.semantics.model.DefaultSemanticTagProvider;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is the main implementing class of the {@link SemanticTagRegistry} interface. It
+ * keeps track of all declared semantic tags of all semantic tags providers and keeps
+ * their current state in memory.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+@NonNullByDefault
+@Component(immediate = true)
+public class SemanticTagRegistryImpl extends AbstractRegistry<SemanticTag, String, SemanticTagProvider>
+        implements SemanticTagRegistry {
+
+    private final Logger logger = LoggerFactory.getLogger(SemanticTagRegistryImpl.class);
+
+    private final ManagedSemanticTagProvider managedProvider;
+
+    @Activate
+    public SemanticTagRegistryImpl(@Reference DefaultSemanticTagProvider defaultSemanticTagProvider,
+            @Reference ManagedSemanticTagProvider managedProvider) {
+        super(SemanticTagProvider.class);
+        // The default semantic tag provider is added immediately without waiting for provider tracker
+        // to be sure that it will be added first (that is before the managed provider). It leads to
+        // a warning log when this provider is added later through the provider tracker.
+        addProvider(defaultSemanticTagProvider);
```

Comment:
If you call `super.addProvider` here, create an override for `addProvider` and check if the added provider is the managed or the default provider (and only call `super.addProvider` if that is not the case), you should be able to workaround the warning.

Reply:
I should probably do that only for the default provider.

Note that we will have similar difficulties later when we will have the managed provider and the config file provider, in case there are ancestor relations between each of them !

Reply:
Done and works well.